packages/driver/src/dom/elements/find.ts

```diff
@@ -247,6 +247,10 @@ export const getContainsSelector = (text, filter = '', options: {
 
     // taken from jquery's normal contains method
     cyContainsSelector = function (elem) {
+      if (elem.type === 'submit' && elem.tagName === 'INPUT') {
+        return text.test(elem.value)
+      }
```

Comment:
How can we handle the default submit buttons like `<input type="submit">`?

The default label is up to the user agent. and can be locale-dependent: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/submit#omitting_the_value_attribute

And I have not been able to find a way to retrieve the default submit string... I've tried `.innerText`, I've tried `getComputedStyle`, I've tried `.value` and `.defaultValue`...

If we can't figure out a way to get the default `submit` value to work with `cy.contains`, then I think this change may be unwise, because it won't be immediately clear why `cy.contains` works with some submits and not others. In that case, we should probably recommend users to use `cy.get('input[type=submit][value=whatever]')` if they need to select a submit by value.

Reply:
How about leaving it as a note in [`cy.contains` doc](https://docs.cypress.io/api/commands/contains#Notes)?

It's happening without changes in this PR. Example code below:

```html
<input id="aaa" type="submit" />
```

```js
it('t', () => {
  cy.visit('fixtures/a.html')

  cy.get('#aaa').should('have.value', 'Submit') // fail. value is ''.
})

it('t2', () => {
  cy.visit('fixtures/a.html')

  cy.contains('Submit') // fail
})
```

----

Maybe we can add the explanation like below:

### When `value` is omitted in `<input type=submit>`

The default label is used and can be locale-dependent when the `value` attribute is not specified in `<input type=submit>`. [Check this.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/submit#omitting_the_value_attribute)

`value` attribute is an empty string and there is no programmatic way to retrieve the default label name set by the user agent. It can cause unexpected failure in your tests.

The solution is to:

* compare the empty string (e.g. `cy.get('#el').should('have.value', '')`
* set the `value` attribute (e.g. `<input type=submit value="Submit" />`)

Reply:
Ohhhhhhhhhhhhh, I see what you mean now. Thank you. This is just fixing the existing case for `RegExp`. In that case, it's of course just a `fix` and not breaking or a "new" concern.

In that case, I'll open a PR for the docs, this PR looks good to me

Reply:
https://github.com/cypress-io/cypress-documentation/pull/4608

Reply:
Sounds like we're good to remove the breaking change label then, @flotwig?

Reply:
@rockhold Oh yeah, thanks. Done.